book/extending.qmd

```diff
@@ -30,6 +30,67 @@ The chapter [Create a new learner](#extending-learners) illustrates the steps ne
 
+
+## Creating a ParamSet {#sec-extending-paramset}
+
+Whenever one wants to extend mlr3, one has to define a `r ref("ParamSet")` that defines its configuration space. 
+In this section you will learn how to define such a parameter set.
+As an example, we will reconstruct a subset of the paramset set of a `r ref("LearnerClassifSVM")` that we have already worked with in section TODO.
+
+```{r}
+library("mlr3learners")
+param_set = lrn("classif.svm")$param_set$subset(
+  c("cost", "kernel", "degree", "shrinking", "type"))
+param_set
+```
+
+We can define a parameter set like seen above, by calling the `r ref("ps")` function.
+The function takes named `r ref("Domain")` arguments that are converted into `r ref("Param")`s and creates a `r ref("ParamSet")` from them.
+The purpose of the `r ref("Domain")` object is merely to allow for a more convenient user interface.
+In order to show how a parameter set can be created, we will reconstruct the parameter subset of the `r ref("LearnerClassifSVM",)` we saw earlier.
+
+In order to define this parameter set, we have to pick the right parameter types.
+As of writing this book, there are five domain constructors that produce different parameters when passed to `ps()`.
+
+
+| Constructor               | Description                          | Underlying Class    |
+| :-----------------------: | :----------------------------------: | :-----------------: |
+| `r ref("p_dbl")`          | Real valued parameter ("double")     | `r ref("ParamDbl")` |
+| `r ref("p_int")`          | Integer parameter                    | `r ref("ParamInt")` |
+| `r ref("p_fct")`          | Discrete valued parameter ("factor") | `r ref("ParamFct")` |
+| `r ref("p_lgl")`          | Logical / Boolean parameter          | `r ref("ParamLgl")` |
+| `r ref("p_uty")`          | Untyped parameter                    | `r ref("ParamUty")` |
+
+: `r ref("Domain")` Constructors and their resulting `r ref("Param")`. {#tbl-paradox-define}
+
+The most important arguments for these domain constructors allow to specify:
+
+1. the admissible set of values for the parameter; for example `lower` and `upper` for `r ref("p_int")` and `r ref("p_dbl")` or `levels` for `r ref("p_fct")`. 
+The `r ref("p_uty")` even allows to specify a `custom_check` function.
+1. dependencies between parameters via the argument `depends`.
+   It is an expression that must involve other parameters and be of the form `<param> == <scalar>`, `<param> %in% <vector>`, or multiple of these chained by `&&`.
+1. the `default` value that allows to record the behaviour if no value is set.
+1. parameter transformations via a `trafo`. These will be covered in @sec-paradox-tune that deals with tuning spaces.
+1. `tags` allow to organize parameters. An important example are the `"train"` and `"predict"` tags that indicate in which phase a parameter is used. These will be omitted here and are covered in @sec-extending-learners that explains how to write custom learners.
```

Comment:
Bounds are discussed in Tuning chapter, I believe so are dependencies? The only relevant thing to this chapter is discussion of 'tags' but this definitely does not need its own section, can just go in extending learners